exchange/accounts/accounts.go

```diff
@@ -0,0 +1,303 @@
+			if assetType != asset.All && assetType != subAcctKey.Asset {
+				continue
+			}
+			subAccts = append(subAccts, &SubAccount{
+				ID:        subAcctKey.SubAccount,
+				AssetType: subAcctKey.Asset,
+				Balances:  balances.Public(),
+			})
+		}
+	}
+
+	if len(subAccts) == 0 {
+		return nil, fmt.Errorf("%w for %s credentials %s asset %s", ErrNoSubAccounts, a.Exchange.GetName(), creds, assetType)
+	}
+	return subAccts, nil
+}
+
+// GetBalance returns a copy of the balance for that asset item
+func (a *Accounts) GetBalance(subAccount string, creds *Credentials, aType asset.Item, c currency.Code) (Balance, error) {
+	if err := common.NilGuard(a); err != nil {
+		return Balance{}, err
+	}
+	if !aType.IsValid() {
+		return Balance{}, fmt.Errorf("cannot get balance: %w: %q", asset.ErrNotSupported, aType)
+	}
+
+	if creds.IsEmpty() {
+		return Balance{}, fmt.Errorf("cannot get balance: %w", errCredentialsEmpty)
+	}
+
+	if c.IsEmpty() {
+		return Balance{}, fmt.Errorf("cannot get balance: %w", currency.ErrCurrencyCodeEmpty)
+	}
+
+	a.mu.RLock()
+	defer a.mu.RUnlock()
+
+	subAccts, ok := a.subAccounts[*creds]
+	if !ok {
+		return Balance{}, fmt.Errorf("%w for %s", ErrNoBalances, creds)
+	}
+
+	assets, ok := subAccts[key.SubAccountAsset{
+		SubAccount: subAccount,
+		Asset:      aType,
+	}]
+	if !ok {
+		return Balance{}, fmt.Errorf("%w for %s SubAccount %q %s", ErrNoBalances, a.Exchange.GetName(), subAccount, aType)
+	}
+	b, ok := assets[c.Item]
+	if !ok {
+		return Balance{}, fmt.Errorf("%w for %s SubAccount %q %s %s", ErrNoBalances, a.Exchange.GetName(), subAccount, aType, c)
+	}
+	return b.Balance(), nil
+}
+
+// Save updates the account balances
+// If isSnapshot is true any missing currencies will be removed
+// Credentials will be retrieved from ctx, Use DeployCredentialsToContext.
+// Changes to balances are published individually
+func (a *Accounts) Save(ctx context.Context, subAccts SubAccounts, isSnapshot bool) error {
+	if err := common.NilGuard(a); err != nil {
+		return fmt.Errorf("cannot save holdings: %w", err)
+	}
+	if err := common.NilGuard(a.subAccounts); err != nil {
+		return fmt.Errorf("cannot save holdings: %w", err)
+	}
+
+	creds, err := a.Exchange.GetCredentials(ctx)
+	if err != nil {
+		return err
+	}
+	if creds.IsEmpty() {
+		return fmt.Errorf("%w: %w", errUpdatingBalance, errCredentialsEmpty)
+	}
+
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	var errs error
+	for _, s := range subAccts {
+		if !s.AssetType.IsValid() {
+			errs = common.AppendError(errs, fmt.Errorf("error loading %s[%s] SubAccount holdings: %w", s.ID, s.AssetType, asset.ErrNotSupported))
+			continue
+		}
+
+		accBalances := a.currencyBalances(creds, s.ID, s.AssetType)
+
+		updated := false
+		missing := maps.Clone(accBalances)
+		for curr, newBal := range s.Balances {
+			delete(missing, curr.Item)
+			if newBal.UpdatedAt.IsZero() {
+				newBal.UpdatedAt = time.Now()
+			}
+			if newBal.Currency == currency.EMPTYCODE {
```

Comment:
```suggestion
			if newBal.Currency.IsEmpty() {
```
Is the most minor of all things

Reply:
Fixed https://github.com/gbjk/gocryptotrader/commit/449a8bccf